src/main/java/me/moodcat/database/controllers/RoomDAO.java

```diff
@@ -0,0 +1,43 @@
+package me.moodcat.database.controllers;
+
+import static me.moodcat.database.entities.QRoom.room;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.transaction.Transactional;
+
+import me.moodcat.database.entities.Room;
+
+import com.google.inject.Inject;
+
+/**
+ * The DAO for rooms.
+ *
+ * @author Jaap Heijligers
+ */
+public class RoomDAO extends AbstractDAO<Room> {
+
+    @Inject
+    public RoomDAO(final EntityManager entityManager) {
+        super(entityManager);
+    }
+
+    /**
+     * Find a room by id.
+     * 
+     * @param id
+     *            the id of the room
+     * @return the {@link Room}
+     */
+    @Transactional
+    public Room findById(final int id) {
+        return this.query().from(room)
```

Comment:
We might want to do an is null check here. I suggest creating a method in `AbstractDAO`:
```java
protected T ensureExists(T entity) {
    if(entity == null) {
        throw new EntityNotFoundException();
    }
    return entity;
}
```

then in `findById(id)` we do:
```java
public Room findById(final int id) {
        return ensureExists(this.query().from(room)
                .where(room.id.eq(id))
                .singleResult(room));
}
```

This way we receive proper error messages when a room does not exists. Yesterday we've already lost some time on NPE's by checking against not found rooms. Throwing an `EntityNotFoundException` consistently also allow us to intercept the exception and send a proper 404 not found response.